sov-rollup-starter/src/rollup.rs

```diff
@@ -1,110 +1,99 @@
 //! Defines the rollup full node implementation, including logic for configuring
 //! and starting the rollup node.
 
-use jsonrpsee::RpcModule;
-use serde::de::DeserializeOwned;
+use std::path::PathBuf;
+
+use async_trait::async_trait;
 use sov_db::ledger_db::LedgerDB;
 use sov_modules_api::default_context::{DefaultContext, ZkDefaultContext};
 use sov_modules_api::Spec;
-use sov_modules_stf_template::{AppTemplate, SequencerOutcome, TxEffect};
+use sov_modules_rollup_template::RollupTemplate;
+use sov_modules_stf_template::{Runtime as RuntimeTrait, SequencerOutcome, TxEffect};
+use sov_risc0_adapter::host::Risc0Host;
+use sov_rollup_interface::mocks::{MockDaConfig, MockDaService, MockDaSpec};
 use sov_rollup_interface::services::da::DaService;
-use sov_rollup_interface::zk::ZkvmHost;
-use sov_state::storage::NativeStorage;
-use sov_stf_runner::{Prover, RollupConfig, RunnerConfig, StateTransitionRunner};
-use stf_starter::{get_rpc_methods, GenesisConfig, Runtime, StfWithBuilder};
-use tokio::sync::oneshot;
-
-use crate::register_rpc::register_sequencer;
-type ZkStf<Da, Vm> = AppTemplate<ZkDefaultContext, Da, Vm, Runtime<ZkDefaultContext, Da>>;
-
-/// Dependencies needed to run the rollup.
-/// This is duplicated exactly from demo-rollup. Should go to stf-runner crate?
-pub struct Rollup<Vm: ZkvmHost, Da: DaService + Clone> {
-    // Implementation of the STF.
-    pub(crate) app: StfWithBuilder<Vm, Da::Spec>,
-    // Data availability service.
-    pub(crate) da_service: Da,
-    // Ledger db.
-    pub(crate) ledger_db: LedgerDB,
-    // Runner configuration.
-    pub(crate) runner_config: RunnerConfig,
-    // Initial rollup configuration.
-    pub(crate) genesis_config: GenesisConfig<DefaultContext, Da::Spec>,
-    // Prover for the rollup.
-    #[allow(clippy::type_complexity)]
-    pub(crate) prover: Option<Prover<ZkStf<Da::Spec, Vm::Guest>, Da, Vm>>,
-}
+use sov_state::config::Config as StorageConfig;
+use sov_state::{ProverStorage, Storage, ZkStorage};
+use sov_stf_runner::RollupConfig;
+use stf_starter::genesis_config::{get_genesis_config, GenesisPaths};
+use stf_starter::{get_rpc_methods, Runtime};
+
+use crate::register_sequencer::register_sequencer;
+
+/// Rollup with MockDa.
+pub struct StarterRollup {}
+
+#[async_trait]
+impl RollupTemplate for StarterRollup {
+    type DaService = MockDaService;
+    type GenesisPaths = GenesisPaths<PathBuf>;
+    type Vm = Risc0Host<'static>;
```

Comment:
Any reason we are coupling it to `Risc0`? Might be worthy to comment here

Reply:
Yes, we have `Rollup` per `DA` and `VM `so the `DaService` & `Vm` needs to be hardcoded at this point.